local.tf

```diff
@@ -144,10 +144,10 @@ locals {
 
   kubeconfig = var.create_eks ? templatefile("${path.module}/templates/kubeconfig.tpl", {
     kubeconfig_name                   = local.kubeconfig_name
-    endpoint                          = aws_eks_cluster.this[0].endpoint
-    cluster_auth_base64               = aws_eks_cluster.this[0].certificate_authority[0].data
+    endpoint                          = length(aws_eks_cluster.this) > 0 ? aws_eks_cluster.this[0].endpoint : ""
+    cluster_auth_base64               = length(aws_eks_cluster.this) > 0 ? aws_eks_cluster.this[0].certificate_authority[0].data : ""
     aws_authenticator_command         = var.kubeconfig_aws_authenticator_command
-    aws_authenticator_command_args    = length(var.kubeconfig_aws_authenticator_command_args) > 0 ? var.kubeconfig_aws_authenticator_command_args : ["token", "-i", aws_eks_cluster.this[0].name]
+    aws_authenticator_command_args    = length(var.kubeconfig_aws_authenticator_command_args) > 0 ? var.kubeconfig_aws_authenticator_command_args : ["token", "-i", var.cluster_name]
```

Comment:
`coalescelist(aws_eks_cluster.this[*].name, [""])[0]` This is true for all your change.